modules/tokens/public.ts

```diff
@@ -0,0 +1,42 @@
+function randomSelection<T>(
+    count: number,
+    possibilities: T[],
+) {
+    const buffer = crypto.getRandomValues(new Uint32Array(count));
+
+    const output = [] as T[];
+    for (let i = 0; i < count; i++) {
+        output.push(possibilities[buffer[i] % possibilities.length]);
+    }
+
+    return output;
+}
+
+export function generateRandomCodeSecure(
```

Comment:
neat. we need ot refactor tokens itself to use this function since it still depends on an insecure version.